### PR TITLE
Add support for downloading default FT model and embeddings.

### PR DESCRIPTION
**Pull request by keyurva**
* Added a mode to the `build_custom_dc_embeddings.py` script to download default FT model and embeddings.
* Updated `Dockerfile` to call this script when building the docker image.
* Server startup with NL enabled improved from O(mins) to O(seconds).
* Does not download place summaries for custom DCs (to circumvent a GCP auth failure) for now.

### CODE REVIEW

**Comment by keyurva** [comment]
~ make sure we'll be using the appropriate/latest base model+embeddings version?

This does not change the version the custom DC image would be using. The embeddings.yaml is already burnt into the image. Even prior to this change, that was the version used by the custom DC NL server ([go/custom-dc-embeddings](http://goto.google.com/custom-dc-embeddings)). That continues to be the case after this change as well. The only difference is that the model and embeddings are burnt into the image so they don't need to be downloaded on every docker run.

File: shared/lib/gcs.py [hunk 1]
```diff
@@ -26,13 +27,18 @@
 def download_file(bucket: str,
                   filename: str,
                   use_anonymous_client: bool = False) -> str:
+  local_embeddings_path = os.path.join(TEMP_DIR, filename)
+  if os.path.exists(local_embeddings_path):
+    logging.info("File already exists, skipping GCS download: %s",
```
[the remark this answers is not in the document]
Comment (keyurva) [reply shared/lib/gcs.py]: Done.

File: tools/nl/embeddings/build_custom_dc_embeddings.md [hunk 2]
```diff
@@ -28,6 +28,17 @@ To use a different model version, specify the `--model-version` flag.
 --output_dir=/tmp
 ```
 
+## Download default FT model and embeddings
```
[the remark this answers is not in the document]
Comment (keyurva) [reply tools/nl/embeddings/build_custom_dc_embeddings.md]: Done.

File: shared/lib/gcs.py [hunk 3]
```diff
@@ -26,13 +27,18 @@
 def download_file(bucket: str,
                   filename: str,
                   use_anonymous_client: bool = False) -> str:
+  local_embeddings_path = os.path.join(TEMP_DIR, filename)
+  if os.path.exists(local_embeddings_path):
+    logging.info("File already exists: %s, skipping GCS download.",
+                 local_embeddings_path)
+    return local_embeddings_path
+
```
Comment (pradh) [inline shared/lib/gcs.py:35]: Looks like this is a general lib used by many callers?  To be sure, do we want to do this check just in the callsite?

nit:  Maybe rename "local_embeddings_path" to "local_file_path" ?

**Comment by pradh** [comment]
> ~ make sure we'll be using the appropriate/latest base model+embeddings version?
> 
> This does not change the version the custom DC image would be using. The embeddings.yaml is already burnt into the image. Even prior to this change, that was the version used by the custom DC NL server ([go/custom-dc-embeddings](http://goto.google.com/custom-dc-embeddings)). That continues to be the case after this change as well. The only difference is that the model and embeddings are burnt into the image so they don't need to be downloaded on every docker run.

Got it!

**Review by pradh — APPROVED** [review]

**Review by juliawu — APPROVED** [review]
Reviewed the place summary changes, which lgtm. Thanks for the fix!

File: server/__init__.py [hunk 4]
```diff
@@ -287,8 +287,12 @@ def create_app(nl_root=DEFAULT_NL_ROOT):
...
   # Load place explorer summaries & allowlist of places to show summaries for
   # Used when rendering place pages
-  app.config['PLACE_SUMMARY_ALLOW_LIST'] = place_summaries.get_place_allowlist()
-  app.config['PLACE_EXPLORER_SUMMARIES'] = place_summaries.get_place_summaries()
+  app.config[
+      'PLACE_SUMMARY_ALLOW_LIST'] = [] if cfg.CUSTOM else place_summaries.get_place_allowlist(
+      )
+  app.config[
+      'PLACE_EXPLORER_SUMMARIES'] = {} if cfg.CUSTOM else place_summaries.get_place_summaries(
+      )
```
Comment (shifucun) [inline server/__init__.py:295]: nit: shall we have a regular if else block so the if block can hold or CUSTOM handlings so it's easier to update later.

**Review by shifucun — APPROVED** [review]
very nice change and refactor!

File: server/__init__.py (hunk 4 above)
Comment (keyurva) [reply server/__init__.py]: Done.

**Comment by keyurva** [comment]
Thanks all for the reviews!

**Pull request merged**